### PR TITLE
ircii: fix build on older systems and Sonoma

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Fix

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by pmetzger** [comment]
What's the failure on '14?

File: irc/ircii/Portfile [hunk 1]
```diff
@@ -36,6 +36,17 @@ configure.env       COPY_DIRECTORY=tar
...
 
+# translat.h:24: error: redefinition of typedef ‘libiconv_t’
+compiler.blacklist-append \
+                    *gcc-4.0 *gcc-4.2
+
+# help.c:535:50: error: passing argument 3 of 'scandir'
+# from incompatible pointer type [-Wincompatible-pointer-types]
+if {[string match *gcc* ${configure.compiler}]} {
+    configure.cflags-append \
+                    -Wno-error=incompatible-pointer-types
```
Comment (pmetzger) [inline irc/ircii/Portfile:47]: As usual, I note that upstream probably wants to know about such problems.

Comment (barracuda156) [reply irc/ircii/Portfile:47]: Homepage returns “The page you requested could not be found”.

**Comment by barracuda156** [comment]
`clang-18` builds it on Sonoma with no hacks. I will just blacklist Apple clang there.

File: irc/ircii/Portfile (hunk 1 above)
Comment (pmetzger) [reply irc/ircii/Portfile:50]: Try talking to mrg a t netbsd.org

Comment (ryandesign) [reply irc/ircii/Portfile:50]: The current homepage might be http://eterna23.net/ircii/

Comment (barracuda156) [reply irc/ircii/Portfile:50]: “ircII comes with a script called ircbug that should be used to send bug reports”

Sorry, I am not gonna do this. If everyone prefers having this port broken, so be it, it is not important for me. There is more important stuff in the pipeline to complete.

**Comment by pmetzger** [comment]
FYI, I've updated the underlying port, and I've emailed Matt.

**Comment by barracuda156** [comment]
> FYI, I've updated the underlying port, and I've emailed Matt.

Thank you. I rebased to master now. (Both issues with gcc are still there, so fixes are due.)

**Comment by pmetzger** [comment]
Why don't we tell him about the warning? I bet he'll just fix it.

**Comment by reneeotten** [comment]
so, this is still failing. If the issue is only with GCC we could in theory merge this as most people will use Clang anyway. But the CI for macOS 14 is still not passing - what's the path forward?

**Comment by barracuda156** [comment]
> so, this is still failing. If the issue is only with GCC we could in theory merge this as most people will use Clang anyway. But the CI for macOS 14 is still not passing - what's the path forward?

@reneeotten It was building fine with clang-18 on Sonoma. It is an easy fix: avoid a broken Apple clang.

**Comment by reneeotten** [comment]
then please do so...

**Comment by pmetzger** [comment]
If we know exactly what's wrong, I can get Matt to take patches.

**Comment by pmetzger** [comment]
Matt has a new upstream version, I'm going to update to it so that we can have a new default server again.

**Comment by reneeotten** [comment]
poke @pmetzger @barracuda156 let's get this wrapped up plase or close the PR.

**Comment by barracuda156** [comment]
@reneeotten Since apparently @pmetzger wants to do the update, let’s hope he could also borrow compiler blacklist to have older systems fixed.
In the meanwhile I close this.

**Pull request closed without merging**